resources/code-template/java/sendD2C/src/main/java/com/microsoft/docs/iothub/samples/SimulatedDevice.java

```diff
@@ -36,15 +37,16 @@ public String serialize() {
   }
 
   // Print the acknowledgement received from IoT Hub for the telemetry message sent.
-  private static class EventCallback implements IotHubEventCallback {
-    public void execute(IotHubStatusCode status, Object context) {
-      System.out.println("IoT Hub responded to message with status: " + status.name());
-
-      if (context != null) {
-        synchronized (context) {
-          context.notify();
+  private static class EventCallback implements MessageSentCallback {
+    @Override
+    public void onMessageSent(Message sentMessage, IotHubClientException exception, Object callbackContext) {
```

Comment:
Is there not a more specific type for callbackContext or is Object the most appropriate?

Reply:
In this case, the only thing we're using the context object for is to manage the send/receive semaphore - I think an empty object works fine here but honestly I don't remember enough about Java to stand by that decision very strongly :)